### ISSUE TITLE
Upgrade react version in react-jhipster lib

### ISSUE THREAD

**Issue by ctamisier** [jhipster/generator-jhipster#13224]
In react-jhipster lib,
There are some react version still at 16.x.x

I think they have to be upgraded as well to 17.0.1 for the v7 (and maybe some other related dependencies ?)

here: https://github.com/jhipster/react-jhipster/blob/main/package.json

**Comment by pascalgrimaud** [comment]
I don't use React, but I think you're right, the version should be align.
@sendilkumarn @deepu105 : is it correct ?

**Comment by sendilkumarn** [comment]
Yeah we should upgrade to the latest release. Last time when I checked there were a couple of dependencies not working with the latest react version.

**Comment by sendilkumarn** [comment]
@ctamisier would you be interested to do the upgrade?

**Comment by ctamisier** [comment]
I'm starting here: https://github.com/jhipster/react-jhipster/pull/63 (It seems to work for the moment but I will double check everything, and study the use of ~x.y.z and the whole dependency management).

For now, I'm not really confident about the build I make locally and the integration in a local JHipster app. (through `npm link`, `npm link react-jhipster`, ...)
I've some warnings at startup (`npm start`), but this is not due to the upgrade of package but of my "local" build (`npm run build`).
I think my local build is not strictly the same as if it was packaged and released on npmjs.
Is there a way (the easiest as possible), to have a npm tag that matches the `main` branch of `react-jhipster` ?  (this tag could be named `dev` for example ?) (https://docs.npmjs.com/cli/v6/commands/npm-dist-tag#purpose)
This will help to have a preview of the final `generator-jhipster` package that uses the latest `react-jhipster` lib.

And in this case, in `package.json` of `generator-jhipster` we can put:
`"react-jhipster": "dev"` (I'm really not sure it works like this...)

TLDR:
Having a Continuous Deployment for react-jhipster on npmjs (that will make us confident when we'll have to make a release of the `react-jhipster` lib)

**Comment by deepu105** [comment]
Did you try using npm pack and install the generated tar `npm install <tarball
file>`

https://docs.npmjs.com/cli/v6/commands/npm-pack

Thanks & Regards,
Deepu

**Comment by ctamisier** [comment]
Oh yes ! thx for the information !
Actually I should have remembered: https://github.com/jhipster/ng-jhipster/commit/8a8e587ac68c3688f56a72ed93ebc5943f26c7c8#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5

Now the "build warnings" are gone.
I still have some warning about how `dayjs` is integrated and specifically its typescript integration...
I will investigate.

**Comment by pascalgrimaud** [comment]
work is still in progress here: https://github.com/jhipster/react-jhipster/pull/63

**Comment by pascalgrimaud** [comment]
@deepu105 @sendilkumarn : can you help @ctamisier to finish this PR plz ? https://github.com/jhipster/react-jhipster/pull/63

**Comment by DanielFran** [comment]
Closed by #13735
